### PR TITLE
redis 4.0.12

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Now(October 23th, 2018), we can install redis@5 and redis@3 by `brew install`, 
but we cannot do redis@4. 
So, I add a new formula for redis4.

### CODE REVIEW

**Comment by epistrephein** [comment]
Commenting to prevent this from being closed.
I think having redis 4 on homebrew would be very useful for many.

**Comment by epistrephein** [comment]
Commenting again to prevent closing.
Could this get evaluated by a maintainer? In case something is off I can step up and fix it if @hyuraku is not available

File: Formula/redis@4.0.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+class RedisAT40 < Formula
+  desc "Persistent key-value database, with built-in net interface"
+  homepage "https://redis.io/"
+  url "http://download.redis.io/releases/redis-4.0.11.tar.gz"
```
Comment (igas) [inline Formula/redis@4.0.rb]: https://github.com/antirez/redis/releases/tag/4.0.12 released

File: Formula/redis@4.0.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  sha256 "fc53e73ae7586bcdacb4b63875d1ff04f68c5474c1ddeda78f00e5ae2eed1bbb"
+
+  keg_only :versioned_formula
+
+  option "with-jemalloc", "Select jemalloc as memory allocator when building Redis"
```
Comment (igas) [inline Formula/redis@4.0.rb]: Please remove option. See #31510 for the rationale behind removing options from formulas.

File: Formula/redis@4.0.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  def install
+    # Architecture isn't detected correctly on 32bit Snow Leopard without help
+    ENV["OBJARCH"] = "-arch #{MacOS.preferred_arch}"
+
+    args = %W[
+      PREFIX=#{prefix}
+      CC=#{ENV.cc}
+    ]
+    args << "MALLOC=jemalloc" if build.with? "jemalloc"
+    system "make", "install", *args
```
Comment (igas) [inline Formula/redis@4.0.rb]: without option could be simplified to:
```
system "make", "install", "PREFIX=#{prefix}", "CC=#{ENV.cc}"
```

File: Formula/redis@4.0.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  def install
+    # Architecture isn't detected correctly on 32bit Snow Leopard without help
+    ENV["OBJARCH"] = "-arch #{MacOS.preferred_arch}"
+
+    args = %W[
+      PREFIX=#{prefix}
+      CC=#{ENV.cc}
+    ]
+    args << "MALLOC=jemalloc" if build.with? "jemalloc"
```
Comment (igas) [inline Formula/redis@4.0.rb]: don't forget to drop here

Comment (igas) [inline Formula/redis@4.0.rb]: don't forget to drop here

File: Formula/redis@4.0.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      s.gsub! "/var/run/redis.pid", var/"run/redis.pid"
+      s.gsub! "dir ./", "dir #{var}/db/redis/"
+      s.sub!  /^bind .*$/, "bind 127.0.0.1 ::1"
+    end
+
+    etc.install "redis.conf"
+    etc.install "sentinel.conf" => "redis-sentinel.conf"
+  end
+
+  plist_options :manual => "redis-server #{HOMEBREW_PREFIX}/etc/redis.conf"
```
Comment (igas) [inline Formula/redis@4.0.rb]: I believe this one needs to be updated to opt. See [cassandra@2.1](https://github.com/Homebrew/homebrew-core/blob/e0b944b3ccb73fa1aaea99dc94b57423dd278828/Formula/cassandra%402.1.rb#L97) for example.

**Review by igas — CHANGES_REQUESTED** [review]

**Review by igas — APPROVED** [review]

**Comment by igas** [comment]
Thank you @hyuraku!

**Comment by igas** [comment]
Closed via https://github.com/Homebrew/homebrew-core/commit/d98a6feb8cf67b6a9285ed60041936dc45d7b424

**Pull request closed without merging**